mcs/class/corlib/corefx/Interop.GetRandomBytes.Mono.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+
+internal partial class Interop
+{
+	internal static class MonoGetRandomBytesFallback
+	{
+		static object _rngAccess = new object ();
+		static RandomNumberGenerator _rng;
+
+		internal static void GetRandomBytes (byte[] buffer)
+		{
+			lock (_rngAccess) {
+				if (_rng == null)
+					_rng = new RNGCryptoServiceProvider ();
+				_rng.GetBytes (buffer);
+			}
+		}
+
+		internal static unsafe void GetRandomBytes (byte* buffer, int length)
+		{
+			var bytes = new byte[length];
```

Comment:
To avoid creating a new array and copy around, it would be worthwhile to pass the byte* to the RandomNumberGenerator, even if it means adding an internal API

Reply:
Would be / might be - but this is unrelated to this particular issue and it may not be worth the time anyways.